### ISSUE TITLE
Internal server error when public link is used 

### ISSUE THREAD

**Issue by githubDante** [nextcloud/forms#178]
**Which Forms version are you running?** (see apps page)
git [master] - see #166 why
**Nextcloud or ownCloud?:**
Nextcloud
**Nextcloud/ownCloud version:** (see Nextcloud admin page)
17.0.2
### Steps to reproduce
1. Create a form, set the access type to Public.
2. Open the form link from private window.

### Actual behaviour
Internal server error.

### Expected behaviour
The link should provide access to the form.

### Logs

* Log from the admin panel:
```
[index] Error: Exception: Argument 2 passed to OCA\Forms\Db\NotificationMapper::findByUserAndForm() must be of the type string, null given, called in
<removed>/apps/forms/lib/Controller/PageController.php on line 281 at <<closure>>

0. <removed>/lib/private/AppFramework/App.php line 126
   OC\AppFramework\Http\Dispatcher->dispatch(OCA\Forms\Controller\PageController {}, "gotoForm")
1. <removed>/lib/private/AppFramework/oRuting/RouteActionHandler.php line 47
   OC\AppFramework\App::main("OCA\\Forms\\Controller\\PageController", "gotoForm", OC\AppFramework\ ... {}, {hash: "2tk0hQCg     ... "})
2. <<closure>>
   OC\AppFramework\Routing\RouteActionHandler->__invoke({hash: "2tk0hQCg ... "})
3. <removed>/lib/private/Route/Router.php line 297
   undefinedundefinedcall_user_func(OC\AppFramework\ ... {}, {hash: "2tk0hQCg ... "})
4. <removed>/lib/base.php line 997
   OC\Route\Router->match("/apps/forms/form/2tk0hQCgNLExJ9xK")
5. <removed>/index.php line 42
   OC::handleRequest()
GET /index.php/apps/forms/form/2tk0hQCgNLExJ9xK
```
### Reason
userID will remain null in the scenario described above, which in turn causes the reported error.

### Fix
The following patch contains a fix.

[PageController_notifMapper_patch.txt](https://github.com/nextcloud/forms/files/4091661/PageController_notifMapper_patch.txt)

**Comment by schniepp** [comment]
Worked for me! Hope this will get merged! For 1.1.2?

**Comment by jotoeri** [comment]
Solved with #235 .

**Comment by riverboboy** [comment]
How can I change it?
